rskj-core/src/main/java/co/rsk/net/messages/MessageVisitor.java

```diff
@@ -0,0 +1,266 @@
+    }
+
+    public void apply(SkeletonRequestMessage message) {
+        final long startNumber = message.getStartNumber();
+        this.blockProcessor.processSkeletonRequest(sender, message.getId(), startNumber);
+    }
+
+    public void apply(BlockHeadersRequestMessage message) {
+        final byte[] hash = message.getHash();
+        final int count = message.getCount();
+        this.blockProcessor.processBlockHeadersRequest(sender, message.getId(), hash, count);
+    }
+
+    public void apply(BlockHashRequestMessage message) {
+        this.blockProcessor.processBlockHashRequest(sender, message.getId(), message.getHeight());
+    }
+
+    public void apply(BlockHashResponseMessage message) {
+        this.syncProcessor.processBlockHashResponse(sender, message);
+    }
+
+    public void apply(NewBlockHashMessage message) {
+        if (blockProcessor.hasBetterBlockToSync()) {
+            loggerMessageProcess.debug("Message[{}] not processed.", message.getMessageType());
+            return;
+        }
+        this.syncProcessor.processNewBlockHash(sender, message);
+    }
+
+    public void apply(SkeletonResponseMessage message) {
+        this.syncProcessor.processSkeletonResponse(sender, message);
+    }
+
+    public void apply(BlockHeadersResponseMessage message) {
+        this.syncProcessor.processBlockHeadersResponse(sender, message);
+    }
+
+    public void apply(BodyRequestMessage message) {
+        final byte[] hash = message.getBlockHash();
+        this.blockProcessor.processBodyRequest(sender, message.getId(), hash);
+    }
+
+    public void apply(BodyResponseMessage message) {
+        this.syncProcessor.processBodyResponse(sender, message);
+    }
+
+    public void apply(NewBlockHashesMessage message) {
+        blockProcessor.processNewBlockHashesMessage(sender, message);
+    }
+
+    public void apply(TransactionsMessage message) {
+        if (blockProcessor.hasBetterBlockToSync()) {
+            loggerMessageProcess.debug("Message[{}] not processed.", message.getMessageType());
+            return;
+        }
+
+        long start = System.nanoTime();
+        loggerMessageProcess.debug("Tx message about to be process: {}", message.getMessageContentInfo());
+
+        List<Transaction> messageTxs = message.getTransactions();
+        List<Transaction> txs = new LinkedList<>();
+
+        for (Transaction tx : messageTxs) {
+            if (!tx.acceptTransactionSignature(config.getNetworkConstants().getChainId())) {
+                recordEvent(sender, EventType.INVALID_TRANSACTION);
+            } else {
+                txs.add(tx);
+                recordEvent(sender, EventType.VALID_TRANSACTION);
+            }
+        }
+
+        transactionGateway.receiveTransactionsFrom(txs, sender.getPeerNodeID());
+
+        loggerMessageProcess.debug("Tx message process finished after [{}] nano.", System.nanoTime() - start);
+    }
+
+    public void apply(GetBlockHeadersMessage message) {
```

Comment:
We should remove the `GetBlockHeadersMessage` class

Reply:
Yup